### ISSUE TITLE
Fix for Monterey

### ISSUE THREAD

**Issue by charliebruce** [charliebruce/cobble#10]
Neither the test app nor the Unity plugin seem to get scan results. Possibly a new privacy setting or a difference in the way ObjC events happen?

**Comment by charliebruce** [comment]
This might be an issue with Monterey, not with Apple Silicon - getting same results on Intel MBP.

**Comment by charliebruce** [comment]
This also seems to happen in a build, not just in the Editor (on an Intel Monterey machine)

**Comment by charliebruce** [comment]
This also happens in the command-line example app, not just in Unity projects, so that narrows the search...

**Comment by charliebruce** [comment]
Scan results are found as expected when running nRF Connect for Mobile (iOS app) within macOS on Apple Silicon. So it's not a hardware or driver issue, there's something specific to Monterey.

**Comment by charliebruce** [comment]
```
default	16:04:33.995342+0000	bluetoothd	Sending 'session detached' event for session "cobble_mac-central-50542-547"
default	16:04:33.995371+0000	bluetoothd	Erasing session 0x156e76170 from SessionMap for "cobble_mac-central-50542-547"
default	16:04:33.998328+0000	bluetoothd	Scanning stopped successfully.  42 Advertising Events, 0 dropped pkts
```

So the events have been produced by bluetoothd, but not received by the app?

**Comment by charliebruce** [comment]
Looks like this may be related to running `scanForPeripheralsWithServices:nil`. With a non-nil argument, I get events matching the filter, as expected.

**Comment by charliebruce** [comment]
Worked around in 5d9ff796b90c77374a5b6efc15ef38d4575756a7 but keeping this open until I've confirmed that this is a macOS bug.

**Comment by charliebruce** [comment]
Confirmed as a bug, which is fixed in Monterey 12.3 beta 21E5212f. I've left a note in the code documenting this issue, as it's likely that people will continue to use affected versions of macOS for a while.
